client/app/utilities/store.ts

```diff
@@ -65,6 +65,18 @@ export function saveListToStore<M extends WithId, E extends M = M>(
   list.forEach((entity) => saveEntityToStore(store, entity, false));
 }
 
+/**
+ * Saves the given list of detailed entities to the given entity store. If any of the
+ * entities already exist in the store, they will be merged with the existing entities.
+ * This method is meant to be used within the reducers.
+ */
+export function saveDetailedListToStore<M extends WithId, E extends M = M>(
+  store: EntityStore<M, E>,
+  list: E[],
+): void {
+  list.forEach((entity) => saveEntityToStore(store, entity, true));
+}
+
```

Comment:
Added to support storing a store of `<Entity, Entity>`, as we currently only have `<MiniEntity, Entrity>`

Reply:
We may not be needing this since we are reverting everything to MiniEntity